### PR TITLE
CB-12940: [IOS] Use deployment-target from config.xml for platform ve…

### PR DESCRIPTION
**Pull request by igormartsekha**
…rsion in Podfile

### Platforms affected
IOS

### What does this PR do?
It's generate Podfile using `deployment-target` option from config.xml

### What testing has been done on this change?
I runed test but it failed, because API.js need valid path to config.xml file. So I changed test, and set up valid path.

### Checklist
- [x] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [x] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [x] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by shazron** [comment]
We had to pull in new code for the new linter using eslint. Consequently, some files have changed. Can you try rebasing, then running npm run eslint to make sure it's all kosher

**Comment by shazron** [comment]
Not sure yet why the appveyor CI  (Windows) is failing some tests

**Comment by shazron** [comment]
```
Parsing C:\projects\cordova-ios\tests\spec\unit\fixtures\ios-config-xml\ios-config-xml\SampleApp\config.xml failed
{ Error: ENOENT: no such file or directory, open 'C:\projects\cordova-ios\tests\spec\unit\fixtures\ios-config-xml\ios-config-xml\SampleApp\config.xml'
```
Note that it is doubling up on the `ios-config-xml` in the path. I'll have to test in Windows to see what's up.

**Comment by shazron** [comment]
Please run `npm install && npm run eslint`, then fix the eslint issues and commit them.

**Comment by shazron** [comment]
Or you can check the "Allow Edits from Maintainers" checkbox on this pull request.

**Comment by shazron** [comment]
nvm I see that I have access to your PR branch already :)

**Comment by igormartsekha** [comment]
I fixed problem with failed test on appveyor

File: bin/templates/scripts/cordova/lib/Podfile.js [hunk 1]
```diff
@@ -98,8 +98,8 @@ Podfile.prototype.getTemplate = function () {
     var projectName = this.escapeSingleQuotes(this.projectName);
     return util.format(
         '# DO NOT MODIFY -- auto-generated by Apache Cordova\n' +
-        'platform :ios, \'8.0\'\n' +
-        'target \'%s\' do\n' +
+        'platform :ios, \'%s\'\n' +
+        'use_frameworks!\n' +
```
Comment (shazron) [inline bin/templates/scripts/cordova/lib/Podfile.js]: we can't use `use_frameworks!`. See: https://issues.apache.org/jira/browse/CB-11893

Comment (igormartsekha) [inline bin/templates/scripts/cordova/lib/Podfile.js]: It doesn't allow to add swift libraries in Podfile.

**Comment by igormartsekha** [comment]
Hello
Please review my last changes

**Comment by shazron** [comment]
Unfortunately this PR has branched and diverged too much (my fault - I edited in Github and it branched). I cleaned it up into this new PR #335

**Pull request closed without merging**